activesupport/lib/active_support/broadcast_logger.rb

```diff
@@ -202,5 +202,11 @@ def fatal!
       def dispatch(&block)
         @broadcasts.each { |logger| block.call(logger) }
       end
+
+      def method_missing(name, *args)
+        logger = @broadcasts.first
```

Comment:
Not a big fan of this and that's basically how it used to work previously (custom method on the main logger wouldn't propagate to all loggers).

I'd rather iterate over all the loggers in the broadcast that responds to the method, but the problem is for the return value here which would necessarily be an enumerable, this would be confusing and also breaks a bunch of tests (see the snippet for example)

Decided to leave it as how it used to work

```ruby
class MyCustomLogger < Logger
 def foo
   true
 end
end

# config/application.rb
config.logger = MyCustomLogger.new

# Rails.logger becomes a BroadcastLogger during [initialization](https://github.com/rails/rails/pull/49417/files#diff-55bc5dde25c29798e0e8fceb70e7391a67a262e0cc2550d179eefa0a3bbc4cffR61)
Rails.logger.foo # => [true]
```

Reply:
In that case maybe we should remove the support to passing multiple loggers to the initializer since now we are making one logger special.

Reply:
I figured an alternative that I think is a good compromise: 

Return all returned values by the loggers as an array, or just the raw return value when a single logger in the broadcast could handle the method.

This should allow existing applications who have a custom logger to behave the same as before. For apps or libraries that will use the `BroadcastLogger`, it should be less confusing than having only one logger able to handle custom methods.

```ruby
class MyLogger < Logger
  def foo
    true
  end
end

broadcast = BroadcastLogger.new(MyLogger.new)
broadcast.foo # => true

broadcast.broadcast_to(MyLogger.new)
broadcast.foo # => [true, true]
```

Let me know if that's ok, otherwise I'll revert and apply your suggestions. Thanks 🙇‍♂️